biz.aQute.repository/src/aQute/bnd/repository/maven/pom/provider/BndPomRepository.java

```diff
@@ -88,13 +99,57 @@ public synchronized void init() {
 				repository.close();
 				throw new IllegalStateException("We have neither a pom, revision, or query set!");
 			}
+
 			bridge = new BridgeRepository(repoImpl);
 			inited = true;
 		} catch (Exception e) {
 			throw Exceptions.duck(e);
 		}
 	}
 
+	private void startPoll() {
+		Workspace ws = registry.getPlugin(Workspace.class);
+		if ((ws != null) && (ws.getGestalt().containsKey(Constants.GESTALT_BATCH)
+				|| ws.getGestalt().containsKey(Constants.GESTALT_CI)
+				|| ws.getGestalt().containsKey(Constants.GESTALT_OFFLINE))) {
+			return;
+		}
+		final AtomicBoolean busy = new AtomicBoolean();
+		pomPoller = Processor.getScheduledExecutor().scheduleAtFixedRate(new Runnable() {
+
+			@Override
+			public void run() {
+				if (busy.getAndSet(true))
+					return;
+				try {
+					boolean needsUpdate = false;
+					for (Entry<URI,Long> pomFilesEntry : pomFiles.entrySet()) {
+						if (pomFilesEntry.getValue() == null) {
+							// not interested
+							continue;
+						}
+						File f = new File(pomFilesEntry.getKey());
```

Comment:
The File constructor will throw an exception if the URI is not a file URI. Since non-file URIs are supported, using them should not result in this Runnable constantly reporting errors.